spec/jwt_spec.rb

```diff
@@ -358,12 +384,6 @@
     end
   end
 
-  context 'a token with two segments but does not require verifying' do
-    it 'raises something else than "Not enough or too many segments"' do
-      expect { JWT.decode('ThisIsNotAValidJWTToken.second', nil, false) }.to raise_error(JWT::DecodeError, 'Invalid segment encoding')
-    end
-  end
-
```

Comment:
See the above comments regarding segments count.

There existed circumstances where 2-segment token would be processed without an exception (`"alg": "none"` along with `verify=false`) while such a format is not defined by any RFC and should be invalid.

If this is considered a breaking change we may handle it separately from this PR.